src/libpmemobj/palloc.c

```diff
@@ -269,13 +269,13 @@ palloc_heap_action_exec(struct palloc_heap *heap,
 	/* suppress unused-parameter errors */
 	SUPPRESS_UNUSED(heap);
 
-#ifdef DEBUG
```

Comment:
You add not only logging but also reading. The thought is that this can affect performance.

Reply:
This will cause one extra PMem memory access on all heap operations. It might be worth checking if the performance impact isn't negligible.